# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: docshell/base/BrowsingContext.cpp
# Commit: c11ce752a12e
# Full Hash: c11ce752a12ef6db0e2422858fe2b00471a9ed9b
# Author: Yaron Tausky <ytausky@mozilla.com>
# Date: 2019-12-11 03:34:51
# Regressor Bug: 1593246
# File Overlap Count: 1
# Description:
#   Bug 1593246 - Part 2: Give SessionStorageManager a reference to BrowsingContext r=sg
#   
#   This reference is necessary when sending session storage data for
#   all browsing context to the parent process. Note that it entails
#   making SessionStorageManager a cycle collection participant, since
# ==============================================================================

diff -r 433fd567ff92 -r c11ce752a12e docshell/base/BrowsingContext.cpp
--- a/docshell/base/BrowsingContext.cpp	Tue Dec 10 12:12:25 2019 +0000
+++ b/docshell/base/BrowsingContext.cpp	Tue Dec 10 12:12:32 2019 +0000
@@ -679,7 +679,7 @@
 RefPtr<SessionStorageManager> BrowsingContext::GetSessionStorageManager() {
   RefPtr<SessionStorageManager>& manager = Top()->mSessionStorageManager;
   if (!manager) {
-    manager = new SessionStorageManager();
+    manager = new SessionStorageManager(this);
   }
   return manager;
 }